casper/src/main/rholang/MakePoS.rho

```diff
@@ -0,0 +1,111 @@
+contract @"MakePoS"(@purse, @minimumBond, @maximumBond, @initBonds, return) = {
+  new this, bondsCh, clonePurse, depositPurse in {
+    bondsCh!(initBonds) |
+  
+    //The purpose of this contract is to move the contents
+    //of an existing purse into a new purse. This prevents
+    //the owner of the given purse from modifying the balance
+    //during processing.
+    contract clonePurse(@givenPurse, return) = {
+      new balanceCh, resultCh in {
+        @[givenPurse, "getBalance"]!(*balanceCh) |
+        for(@balance <- balanceCh) {
+          @[givenPurse, "split"]!(balance, *resultCh) |
+          for(@result <- resultCh) {
+            match result {
+              [] => { return!(("Left", "Could not extract funds from given purse")) }
+              [newPurse] => { return!(("Right", (newPurse, balance))) }
```

Comment:
Do we not have to check that the newPurse is from the `thisMint` of the Rev mint?

Reply:
That is a good point. A PoS should be tied to a particular `mint`. We would check the given purse and then just know that the new purse is of the same mint because that is how purses work.